dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
What a discussion. @fuchsto It is still not clear to me, why large structs cannot be wrapped with `dash::Atomic`. As you mentioned, atomic refers to the semantics, but not to the implementation. So if neglecting performance, atomic operations on data of any size should be possible by using a `lock()` and `unlock()`. 

If I get the proposed implementation right, we are going to support only types up to `std::size_t`. Why that?